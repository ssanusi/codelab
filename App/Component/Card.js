import React from 'react';
import {
  View, Text, StyleSheet, Image, TouchableOpacity,
} from 'react-native';
import { secondary } from '../lib/constants';


const Card = ({
  fullName, avatarUrl, username, navigation,
}) => (
  <TouchableOpacity
    style={style.container}
    onPress={() => navigation.navigate('Profile', { username })}
  >
    <View style={style.flexLayout}>
      <Image style={style.thumbnail} source={{ uri: avatarUrl }} />
      <View>
        <Text>{fullName}</Text>
        <View style={style.text}>
          <Text>
            <Image style={style.logo} source={require('../../assets/GitHub-Mark-64px.png')} />
            {' '}
            {username}
          </Text>
        </View>
      </View>
      <View>
        <Image
          style={style.arrowIcon}
          source={require('../../assets/ca96d96471dd4fd694ab44f2d1246918.png')}
        />
      </View>
      <View />
    </View>
  </TouchableOpacity>
);

const style = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    backgroundColor: secondary,
    borderRadius: 6.33,
    width: 331.41,
    height: 83.65,
    marginHorizontal: 18,
    marginTop: 15,
  },
  flexLayout: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
  thumbnail: {
    height: 60,
    width: 60,
    borderRadius: 25,
  },
  arrowIcon: {
    width: 30,
    height: 30,
  },
  logo: {
    height: 12,
    width: 12,
  },
  text: {
    padding: 3,
  },
});

export default Card;
