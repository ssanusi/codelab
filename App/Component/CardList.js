import React from 'react';
import { FlatList, Text } from 'react-native';
import { Query } from 'react-apollo';
import { gql } from 'apollo-boost';
import Card from './Card';

export const GET_USERS = gql`
  query {
    search(query: "location:lagos language:javascript", type: USER, first: 100) {
      edges {
        node {
          ... on User {
            name
            avatarUrl
            login
          }
        }
      }
    }
  }
`;

const CardList = ({ navigation }) => (
  <Query query={GET_USERS}>
    {({ data, loading }) => {
      if (loading) return <Text>Loading...</Text>;
      const { edges } = data.search;
      return (
        <FlatList
          data={edges}
          keyExtractor={(item, index) => index.toString()}
          renderItem={({ item }) => (
            <Card
              fullName={item.node.name}
              avatarUrl={item.node.avatarUrl}
              username={item.node.login}
              navigation={navigation}
            />
          )}
        />
      );
    }}
  </Query>
);

export default CardList;
