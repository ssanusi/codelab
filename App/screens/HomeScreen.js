import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import CardList from '../Component/CardList';
import { mainColor, primary } from '../lib/constants';

const HomeScreen = ({ navigation }) => (
  <View style={style.container}>
    <View style={style.main}>
      <Text style={style.mainText}>Javascript Developers in lagos</Text>
      <CardList navigation={navigation} />
    </View>
  </View>
);

const style = StyleSheet.create({
  container: {
    flex: 1,
  },
  main: {
    backgroundColor: mainColor,
    flex: 1,
  },
  mainText: {
    color: primary,
    fontSize: 10,
    fontWeight: 'bold',
    paddingLeft: 10,
    paddingTop: 10,
    marginBottom: 10,
  },
});

export default HomeScreen;
