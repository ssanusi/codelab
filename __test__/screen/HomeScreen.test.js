import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/test-utils';
import wait from 'waait';
import HomeScreen from '../../App/screens/HomeScreen';
import { GET_USERS } from '../../App/Component/CardList';
import data from '../mockData.json';

it('HomeScreen renders without crashing', async () => {
  const mocks = [
    {
      request: {
        query: GET_USERS,
      },
      result: { data },
    },
  ];
  const renderedHomeScreen = renderer
    .create(
      <MockedProvider mocks={mocks}>
        <HomeScreen />
      </MockedProvider>,
    )
    .toJSON();

  await wait(0);
  expect(renderedHomeScreen).toBeTruthy();
  expect(renderedHomeScreen).toMatchSnapshot();
});
