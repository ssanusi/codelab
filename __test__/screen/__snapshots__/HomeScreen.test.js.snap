// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#E1E2E1",
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#006064",
          "fontSize": 10,
          "fontWeight": "bold",
          "paddingLeft": 10,
          "paddingTop": 10,
        }
      }
    >
      Javascript Developers lagos
    </Text>
    <Text>
      Loading...
    </Text>
  </View>
</View>
`;
